ncep/gdas.py

```diff
@@ -0,0 +1,350 @@
+                    'typeOfLevel': 'isobaricInhPa',
+                    'level': levels,
+                },
+                'r': {
+                    'typeOfLevel': 'isobaricInhPa',
+                    'level': levels,
+                },
+            }
+        }
+
+        grbs = pygrib.open(self.out_filename)
+
+        data = []
+        for level_type, variable_data in variables_to_extract.items():
+            for variable, value in variable_data.items():
+
+                levelType = value['typeOfLevel']
+                desired_level = value['level']
+            
+
+                print(f'Get variable {variable} from file {self.out_filename}:')
+                if level_type == 'surface':
+                    values = get_dataarray(grbs, variable, levelType, desired_level)
+                    data.append(values)
+                else:
+                    for level in desired_level:
+                        values = get_dataarray(grbs, variable, levelType, level)
+                        data.append(values)
+
+                ##units conversion
+                #if variable == 'gh':
+                #    values = values * 9.80665
+
+     
+        with open(f'{self.output_directory}/input_{self.start_datetime.strftime("%Y%m%d%H")}.npy', 'wb') as f:
+            np.save(f, np.array(data))
+        
+
+    def remove_downloaded_data(self):
+        # Remove downloaded data from the specified directory
+        print("Removing downloaded grib2 data...")
+        try:
+            os.system(f"rm -rf {self.local_base_directory}")
+            print("Downloaded data removed.")
+        except Exception as e:
+            print(f"Error removing downloaded data: {str(e)}")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Download and process GDAS data")
+    parser.add_argument("start_datetime", help="Start datetime in the format 'YYYYMMDDHH'")
+    parser.add_argument("-m", "--method", help="method to extact variables from grib2, options: wgrib2, pygrib", default="wgrib2")
+    parser.add_argument("-s", "--source", help="the source repository to download gdas grib2 data, options: nomads (up-to-date), s3", default="s3")
+    parser.add_argument("-o", "--output", help="Output directory for processed data")
+    parser.add_argument("-d", "--download", help="Download directory for raw data")
+    parser.add_argument("-k", "--keep", help="Keep downloaded data (yes or no)", default="no")
+
+    args = parser.parse_args()
+
+    start_datetime = datetime.strptime(args.start_datetime, "%Y%m%d%H")
+    download_source = args.source
+    method = args.method
+    output_directory = args.output
+    download_directory = args.download
+    keep_downloaded_data = args.keep.lower() == "yes"
+
+    data_processor = GFSDataProcessor(start_datetime, download_source, output_directory, download_directory, keep_downloaded_data)
+    data_processor.get_data(args.method)
+    
+    #if method == "wgrib2":
```

Comment:
@LinlinCui-NOAA I was wondering if these needs to be uncommented, if not please remove them.

Reply:
> @LinlinCui-NOAA I was wondering if these needs to be uncommented, if not please remove them.

This has been moved to function `get_data` and no longer needed. I deleted them in the new commit.